### DISCUSSION TITLE
TypeError: Cannot read property 'map' of undefined

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#22794] [Help]
### What version of Next.js are you using?

10.0.7

### What version of Node.js are you using?

15.10.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

when returning data from a ts object into a `products` component, next tells me that my list `productList` is undefined.
here's the code for i've tried: 

```js
export function getInitialProps() {
	return {
		props: {
			productListData: productList
		}
	}
}

const Store = function ({ productListData }, props: any) {
  return (
    <Layout>
			<div className={styles.container}>
				<h1> Our Products </h1>
				<Prices bpi={props.bpi}/>
			</div>
			
			<div className="container">
				{productListData.map(productListDb => (
					<div>
						<Products key={productListDb.id} {...productListDb} />
					</div>
				))}
        
      </div>
    </Layout>
  )
}
```

### Expected Behavior

the `map` function should return the products into components within a grid

### To Reproduce

full code for the page (store/index.tsx)
```js
export function getInitialProps() {
	return {
		props: {
			productListData: productList
		}
	}
}

const Store = function ({ productListData }, props: any) {
  return (
    <Layout>
			<div className={styles.container}>
				<h1> Our Products </h1>
				<Prices bpi={props.bpi}/>
			</div>
			
			<div className="container">
				{productListData.map(productListDb => (
					<div>
						<Products key={productListDb.id} {...productListDb} />
					</div>
				))}
        
      </div>
    </Layout>
  )
}
		
export default Store

Products.getServerSideProps = async function (props: any) {
	const res = await fetch('https://api.coindesk.com/v1/bpi/currentprice.json'),
		data = await res.json()
		
	return {
		bpi: data.bpi
	};
	
}
```

the code used for the ts object; (productlist.ts)
```js
export const productList = 
    [{
        id: 1,
        name: 'product name',
        description: 'stuff here to note about the product',
        price: 15,

    },
}
```

**Comment by ghost** [comment]
and yes i just realised that i have `Products.getInitialProps` instead of `Store...` but the error persists

**Reply by hsnilsson** [reply]
This question is better suited for StackOverflow since it is certainly not a bug in Next.

I'm not able to run your code at the moment, but looks like you're erroneously passing `productListData` as a separate argument to your `Store` component. `productListData` is part of your props, so to receive it propperly into your component, try change `({ productListData }, props: any)` into `(props: any)` and access it by `props.productListData` instead.
